include/seastar/http/httpd.hh

```diff
@@ -87,10 +89,18 @@ class connection : public boost::intrusive::list_base_hook<> {
 public:
     [[deprecated("use connection(http_server&, connected_socket&&)")]]
     connection(http_server& server, connected_socket&& fd,
-            socket_address) : connection(server, std::move(fd)) {}
+            socket_address client_addr)
+            : connection(server, std::move(client_addr), 
+                    fd.local_address(), 
+                    std::move(fd)) {}
     connection(http_server& server, connected_socket&& fd)
+            : connection(server, fd.remote_address(), 
+                    fd.local_address(), std::move(fd)) {}
+    connection(http_server& server, socket_address client_addr,
+            socket_address server_addr, connected_socket&& fd) 
             : _server(server), _fd(std::move(fd)), _read_buf(_fd.input()), _write_buf(
-                    _fd.output()) {
+                    _fd.output()), _client_addr(std::move(client_addr)),
+                    _server_addr(std::move(server_addr)) {
```

Comment:
nit: Please, reformat the initializer list so that it looks like

```c++
    connection(...)
        : _server(server)
        , _fd(std::move(fd))
        , _read_buf(_fd.input())
        ...
```